kotlinx-coroutines-core/jvm/src/internal/LockFreeLinkedList.kt

```diff
@@ -611,11 +625,21 @@ public actual open class LockFreeLinkedListNode {
         }
     }
 
+    internal fun nextValue(): Any = _next.value
+
     internal fun validateNode(prev: Node, next: Node) {
-        assert { prev === this._prev.value }
-        assert { next === this._next.value }
+        val result = validateNodeResult(prev, next)
+        check(result.isEmpty()) { result }
+    }
+
+    internal fun validateNodeResult(prev: Node, next: Node): String = buildString {
+        append(validateSameResult("prev", prev, _prev.value))
```

Comment:
It seems that `next` property completes the descriptor stored in it, and this is the only way `prev` and `_prev.value` to be different. However, reading `prev` at first always completes the descriptor. Thus, it seems that this check is useless. I would avoid modifications in such validation functions, it is error-prone.

Reply:
It seems that I've fixed it -- I'm checking raw `_next.value`.

Reply:
Aaah, I see -- you use the same names for the argument and the field, it confused me a bit. Then it is correct, you are right.